### PR TITLE
[DS-4486] Tests for the Comm/Coll admins group edits

### PR DESCRIPTION
**Pull request by KevinVdV**
## References
* [JIRA](https://jira.lyrasis.org/browse/DS-4486)

## Description
This PR includes tests for the edits of the community / collection roles by community / collection admin.

This PR does NOT include a fix where community/collection admins can search for groups/epeople to add.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [X] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide)
- [X] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [X] My PR passes all tests and includes new/updated Unit or Integration Tests for any bug fixes, improvements or new features. A few reminders about what constitutes good tests:
    * Include tests for different user types, including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for known error scenarios and error codes (e.g. `400 Bad Request`, `401 Unauthorized`, `403 Forbidden`, `404 Not Found`, etc)
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
- [X] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [X] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 1]
```diff
@@ -0,0 +1,919 @@
...
+    public void setup() {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context).withName("test").build();
+        collection = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void communityAdminAddMembersToCommunityAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:71]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L2108

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+        getClient(token).perform(get("/api/eperson/groups/" + adminGroup.getID() + "/epersons"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.epersons", Matchers.hasItem(
+                            EPersonMatcher.matchEPersonOnEmail(ePerson.getEmail())
+                        )));
+
+    }
+
+    @Test
+    public void communityAdminRemoveMembersFromCommunityAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:97]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L2108

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+        getClient(token).perform(get("/api/eperson/groups/" + adminGroup.getID() + "/epersons"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.epersons", Matchers.not(Matchers.hasItem(
+                            EPersonMatcher.matchEPersonOnEmail(ePerson.getEmail())
+                        ))));
+
+    }
+
+    @Test
+    public void communityAdminAddChildGroupToCommunityAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:132]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L626

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 4]
```diff
@@ -0,0 +1,919 @@
...
+        getClient(token).perform(get("/api/eperson/groups/" + adminGroup.getID() + "/subgroups"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.subgroups", Matchers.hasItem(
+                            GroupMatcher.matchGroupWithName(group.getName())
+                        )));
+
+    }
+
+    @Test
+    public void communityAdminRemoveChildGroupFromCommunityAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:158]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L1236

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+    }
+
+
+
+
+
+
+
+    @Test
+    public void communityAdminAddMembersToCollectionAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:351]: I will rename my test to communityAdminCanManageCollectionSubmittersGroupAndAdminsGroupsTest and add 
this test inside my test 
https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L2323

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 6]
```diff
@@ -0,0 +1,919 @@
...
+        getClient(token).perform(get("/api/eperson/groups/" + adminGroup.getID() + "/epersons"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.epersons", Matchers.hasItem(
+                            EPersonMatcher.matchEPersonOnEmail(ePerson.getEmail())
+                        )));
+
+    }
+
+    @Test
+    public void communityAdminRemoveMembersFromCollectionAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:377]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L2323

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java [hunk 7]
```diff
@@ -0,0 +1,919 @@
...
+    }
+
+
+
+
+
+
+
+    @Test
+    public void collectionAdminAddMembersToCollectionAdminGroupSuccess() throws Exception {
```
Comment (Micheleboychuk) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionPermissionIT.java:639]: https://github.com/DSpace/DSpace/blob/34b61f3b98eb22436a652f6270ff32e85cd6eb2f/dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java#L2159

**Comment by abollini** [comment]
hi @KevinVdV 
@Micheleboychuk has integrated your test in the PR #2774 do you agree about closing this PR and maybe do you like to review the "full PR" by michele?

**Comment by tdonohue** [comment]
Closing this PR as it has be integrated into #2774.

**Pull request closed without merging**